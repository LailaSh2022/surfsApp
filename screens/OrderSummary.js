import React, { useState } from "react";
import { StyleSheet, View, Text } from "react-native";
import {
  StyledContainer,
  StyledBackButton,
  ButtonText,
  StyledButton,
} from "./../components/Styles";
import "../global";
import { getLastOrderId, InsertOrder } from "../Database";
import { useRoute, useNavigation } from "@react-navigation/native";

const OrderSummary = ({ route }) => {
  const navigation = useNavigation();
  const [orderNumber, setOrderNumber] = useState(0);

  // Updated by Laila Shihada to get the data from the global.js file
  console.log(global.ReceiverId[0]);
  const currentDate = new Date();
  const currentYear = currentDate.getFullYear();
  const currentMonth = currentDate.getMonth() + 1; // January is 0, so add 1 to get the actual month number
  const currentDay = currentDate.getDate();
  const date = currentMonth + "/" + currentDay + "/" + currentYear;

  getLastOrderId()
    .then((lastOrderId) => {
      setOrderNumber(lastOrderId + 1);
    })
    .catch((error) => {
      console.error(`Error retrieving last order id: ${error}`);
    });
  console.log(global.userId[0]);
  const continueHandler = () => {
    InsertOrder({
      SenderId: global.userId[0],
      RecipientId: global.ReceiverId[0],
      Amount: global.FromAmount[0],
      Fee: global.fee[0],
      Rate: global.rate[0],
      FromCurrency: global.FromCurrency[0],
      ToCurrency: global.ToCurrency[0],
      Date: date,
    })
      .then(() => console.log("success"))
      .catch((error) => console.log("error"));
    navigation.navigate("History", { userId: global.userId[0] });
  };

  return (
    <StyledContainer>
      <View style={{ flexDirection: "row" }}>
        <StyledBackButton>
          <ButtonText>{"<"}</ButtonText>
        </StyledBackButton>
        <Text style={styles.headline}> Order Summary</Text>
      </View>
      <View style={{ flexDirection: "row", marginTop: 30 }}>
        <Text style={{ width: "25%" }}>Order No:</Text>
        <Text>{orderNumber}</Text>
      </View>
      <View style={{ flexDirection: "row", marginTop: 15 }}>
        <Text style={{ width: "25%" }}>Date:</Text>
        <Text>
          {currentMonth}
          {"/"}
          {currentDay}
          {"/"}
          {currentYear}
        </Text>
      </View>
      <View style={styles.line} />
      <View style={{ flexDirection: "row", marginTop: 15 }}>
        <Text style={{ width: "25%" }}>Rate:</Text>
        <Text style={{ fontSize: 20, marginTop: -5 }}>
          1.00{global.FromCurrency[0]} = {global.rate[0]}
          {global.ToCurrency[0]}
        </Text>
      </View>
      <View style={styles.line} />
      <View style={{ flexDirection: "row", marginTop: 15 }}>
        <Text>Transfer Info:</Text>
      </View>
      <View style={{ flexDirection: "row", marginTop: 15 }}>
        <Text style={{ width: "25%" }}>Amount:</Text>
        <Text style={{ fontSize: 20, marginTop: -5 }}>
          {global.FromAmount[0]}
          {global.FromCurrency[0]}
        </Text>
      </View>

      <View style={{ flexDirection: "row", marginTop: 15 }}>
        <Text style={{ width: "25%" }}>Fee (1%):</Text>
        <Text style={{ fontSize: 20, marginTop: -5 }}>
          {global.fee[0]}
          {global.ToCurrency[0]}
        </Text>
      </View>
      <View style={{ flexDirection: "row", marginTop: 15 }}>
        <Text style={{ width: "25%" }}>Total:</Text>
        <Text style={{ fontSize: 20, marginTop: -5 }}>
          {global.Amount[0]}
          {global.ToCurrency[0]}
        </Text>
      </View>
      <View style={styles.line} />
      <View style={{ flexDirection: "row", marginTop: 15 }}>
        <Text style={{ width: "40%" }}>Total Receiver gets:</Text>
        <Text style={{ paddingBottom: 30, fontSize: 20, marginTop: -5 }}>
          {global.Amount[0]}
          {global.ToCurrency[0]}
        </Text>
      </View>
      <StyledButton onPress={() => continueHandler()}>
        <ButtonText>Continue</ButtonText>
      </StyledButton>
    </StyledContainer>
  );
};

const styles = StyleSheet.create({
  headline: {
    flex: 2,
    textAlign: "center",
    fontWeight: "bold",
    fontSize: 18,
    paddingTop: 10,
    color: "#152A5F",
  },

  line: {
    paddingTop: 10,
    borderBottomColor: "black",
    borderBottomWidth: StyleSheet.hairlineWidth,
  },
});

export default OrderSummary;
