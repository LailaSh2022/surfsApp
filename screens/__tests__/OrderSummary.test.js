import React from "react";
import renderer from "react-test-renderer";
import OrderSummary from "../OrderSummary";

jest.mock("@react-navigation/native", () => ({
  useNavigation: () => ({ navigate: jest.fn() }),
  useRoute: () => ({ params: {} }),
}));

jest.mock("../../Database", () => ({
  getLastOrderId: () => Promise.resolve(1),
}));

describe("OrderSummary", () => {
  it("renders correctly", async () => {
    const tree = renderer.create(<OrderSummary />).toJSON();
    await new Promise((resolve) => setTimeout(resolve, 100)); // wait for useEffect to complete
    expect(tree).toMatchSnapshot();
  });
});
