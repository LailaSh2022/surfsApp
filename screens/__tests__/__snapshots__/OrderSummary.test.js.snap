// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing OrderSummary screen against its snapshot. If there is unexpected change in UI, it will prompt test failed. 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#ffffff",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "paddingBottom": 25,
        "paddingLeft": 25,
        "paddingRight": 25,
        "paddingTop": 64,
      },
    ]
  }
>
  <View
    style={
      {
        "flexDirection": "row",
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "backgroundColor": "#152A5F",
          "borderRadius": 5,
          "justifyContent": "center",
          "opacity": 1,
          "paddingBottom": 10,
          "paddingLeft": 10,
          "paddingRight": 10,
          "paddingTop": 10,
        }
      }
    >
      <Text
        style={
          [
            {
              "color": "#ffffff",
              "fontSize": 16,
              "textAlign": "center",
            },
          ]
        }
      >
        &lt;
      </Text>
    </View>
    <Text
      style={
        {
          "color": "#152A5F",
          "flex": 2,
          "fontSize": 18,
          "fontWeight": "bold",
          "paddingTop": 10,
          "textAlign": "center",
        }
      }
    >
       Order Summary
    </Text>
  </View>
  <View
    style={
      {
        "flexDirection": "row",
        "marginTop": 30,
      }
    }
  >
    <Text
      style={
        {
          "width": "25%",
        }
      }
    >
      Order No:
    </Text>
    <Text>
      1
    </Text>
  </View>
  <View
    style={
      {
        "flexDirection": "row",
        "marginTop": 15,
      }
    }
  >
    <Text
      style={
        {
          "width": "25%",
        }
      }
    >
      Date:
    </Text>
    <Text>
      16/04/2023
    </Text>
  </View>
  <View
    style={
      {
        "borderBottomColor": "black",
        "borderBottomWidth": 0.5,
        "paddingTop": 10,
      }
    }
  />
  <View
    style={
      {
        "flexDirection": "row",
        "marginTop": 15,
      }
    }
  >
    <Text
      style={
        {
          "width": "25%",
        }
      }
    >
      Rate:
    </Text>
    <Text
      style={
        {
          "fontSize": 20,
          "marginTop": -5,
        }
      }
    >
      1.00
      NZD
       = 
      3.05
      BRL
    </Text>
  </View>
  <View
    style={
      {
        "borderBottomColor": "black",
        "borderBottomWidth": 0.5,
        "paddingTop": 10,
      }
    }
  />
  <View
    style={
      {
        "flexDirection": "row",
        "marginTop": 15,
      }
    }
  >
    <Text>
      Transfer Info:
    </Text>
  </View>
  <View
    style={
      {
        "flexDirection": "row",
        "marginTop": 15,
      }
    }
  >
    <Text
      style={
        {
          "width": "25%",
        }
      }
    >
      Amount:
    </Text>
    <Text
      style={
        {
          "fontSize": 20,
          "marginTop": -5,
        }
      }
    >
      350
      NZD
    </Text>
  </View>
  <View
    style={
      {
        "flexDirection": "row",
        "marginTop": 15,
      }
    }
  >
    <Text
      style={
        {
          "width": "25%",
        }
      }
    >
      Fee (1%):
    </Text>
    <Text
      style={
        {
          "fontSize": 20,
          "marginTop": -5,
        }
      }
    >
      3.50
      NZD
    </Text>
  </View>
  <View
    style={
      {
        "flexDirection": "row",
        "marginTop": 15,
      }
    }
  >
    <Text
      style={
        {
          "width": "25%",
        }
      }
    >
      Total:
    </Text>
    <Text
      style={
        {
          "fontSize": 20,
          "marginTop": -5,
        }
      }
    >
      353.50
      NZD
    </Text>
  </View>
  <View
    style={
      {
        "borderBottomColor": "black",
        "borderBottomWidth": 0.5,
        "paddingTop": 10,
      }
    }
  />
  <View
    style={
      {
        "flexDirection": "row",
        "marginTop": 15,
      }
    }
  >
    <Text
      style={
        {
          "width": "40%",
        }
      }
    >
      Total Receiver gets:
    </Text>
    <Text
      style={
        {
          "fontSize": 20,
          "marginTop": -5,
          "paddingBottom": 30,
        }
      }
    >
      1067.50
      NZD
    </Text>
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "#152A5F",
        "borderRadius": 5,
        "height": 60,
        "justifyContent": "center",
        "marginVertical": 5,
        "opacity": 1,
        "paddingBottom": 15,
        "paddingLeft": 15,
        "paddingRight": 15,
        "paddingTop": 15,
      }
    }
  >
    <Text
      style={
        [
          {
            "color": "#ffffff",
            "fontSize": 16,
            "textAlign": "center",
          },
        ]
      }
    >
      Continue
    </Text>
  </View>
</View>
`;

exports[`UI does not change unexpectedly 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#ffffff",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "paddingBottom": 25,
        "paddingLeft": 25,
        "paddingRight": 25,
        "paddingTop": 64,
      },
    ]
  }
>
  <View
    style={
      {
        "flexDirection": "row",
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "backgroundColor": "#152A5F",
          "borderRadius": 5,
          "justifyContent": "center",
          "opacity": 1,
          "paddingBottom": 10,
          "paddingLeft": 10,
          "paddingRight": 10,
          "paddingTop": 10,
        }
      }
    >
      <Text
        style={
          [
            {
              "color": "#ffffff",
              "fontSize": 16,
              "textAlign": "center",
            },
          ]
        }
      >
        &lt;
      </Text>
    </View>
    <Text
      style={
        {
          "color": "#152A5F",
          "flex": 2,
          "fontSize": 18,
          "fontWeight": "bold",
          "paddingTop": 10,
          "textAlign": "center",
        }
      }
    >
       Order Summary
    </Text>
  </View>
  <View
    style={
      {
        "flexDirection": "row",
        "marginTop": 30,
      }
    }
  >
    <Text
      style={
        {
          "width": "25%",
        }
      }
    >
      Order No:
    </Text>
    <Text>
      1
    </Text>
  </View>
  <View
    style={
      {
        "flexDirection": "row",
        "marginTop": 15,
      }
    }
  >
    <Text
      style={
        {
          "width": "25%",
        }
      }
    >
      Date:
    </Text>
    <Text>
      16/04/2023
    </Text>
  </View>
  <View
    style={
      {
        "borderBottomColor": "black",
        "borderBottomWidth": 0.5,
        "paddingTop": 10,
      }
    }
  />
  <View
    style={
      {
        "flexDirection": "row",
        "marginTop": 15,
      }
    }
  >
    <Text
      style={
        {
          "width": "25%",
        }
      }
    >
      Rate:
    </Text>
    <Text
      style={
        {
          "fontSize": 20,
          "marginTop": -5,
        }
      }
    >
      1.00
      NZD
       = 
      3.05
      BRL
    </Text>
  </View>
  <View
    style={
      {
        "borderBottomColor": "black",
        "borderBottomWidth": 0.5,
        "paddingTop": 10,
      }
    }
  />
  <View
    style={
      {
        "flexDirection": "row",
        "marginTop": 15,
      }
    }
  >
    <Text>
      Transfer Info:
    </Text>
  </View>
  <View
    style={
      {
        "flexDirection": "row",
        "marginTop": 15,
      }
    }
  >
    <Text
      style={
        {
          "width": "25%",
        }
      }
    >
      Amount:
    </Text>
    <Text
      style={
        {
          "fontSize": 20,
          "marginTop": -5,
        }
      }
    >
      350
      NZD
    </Text>
  </View>
  <View
    style={
      {
        "flexDirection": "row",
        "marginTop": 15,
      }
    }
  >
    <Text
      style={
        {
          "width": "25%",
        }
      }
    >
      Fee (1%):
    </Text>
    <Text
      style={
        {
          "fontSize": 20,
          "marginTop": -5,
        }
      }
    >
      3.50
      NZD
    </Text>
  </View>
  <View
    style={
      {
        "flexDirection": "row",
        "marginTop": 15,
      }
    }
  >
    <Text
      style={
        {
          "width": "25%",
        }
      }
    >
      Total:
    </Text>
    <Text
      style={
        {
          "fontSize": 20,
          "marginTop": -5,
        }
      }
    >
      353.50
      NZD
    </Text>
  </View>
  <View
    style={
      {
        "borderBottomColor": "black",
        "borderBottomWidth": 0.5,
        "paddingTop": 10,
      }
    }
  />
  <View
    style={
      {
        "flexDirection": "row",
        "marginTop": 15,
      }
    }
  >
    <Text
      style={
        {
          "width": "40%",
        }
      }
    >
      Total Receiver gets:
    </Text>
    <Text
      style={
        {
          "fontSize": 20,
          "marginTop": -5,
          "paddingBottom": 30,
        }
      }
    >
      1067.50
      NZD
    </Text>
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "#152A5F",
        "borderRadius": 5,
        "height": 60,
        "justifyContent": "center",
        "marginVertical": 5,
        "opacity": 1,
        "paddingBottom": 15,
        "paddingLeft": 15,
        "paddingRight": 15,
        "paddingTop": 15,
      }
    }
  >
    <Text
      style={
        [
          {
            "color": "#ffffff",
            "fontSize": 16,
            "textAlign": "center",
          },
        ]
      }
    >
      Continue
    </Text>
  </View>
</View>
`;
